### PR TITLE
Prototype of possible VA1 dataplane kustomization

### PR DESCRIPTION
**Pull request by abays**
Changes VA1 to use `kustomize` for both control plane and data plane.  If we use this approach, then the pattern for using any given VA would be:

1. Go to a certain dir, i.e. `examples/va/<some va>`
2. Edit a `values.yaml`
3. `kustomize build` that dir
4. `oc apply` the resultant YAML
5. Repeat 0..N times in appropriate dirs in that `examples/va/<some va>` dir until all stages of the particular VA are completed

### CODE REVIEW

File: examples/va/hci/values.yaml
Comment (fultonj) [inline examples/va/hci/values.yaml:17]: I think it's fine to keep this `hci/values.yaml` as it, but I wanted to point out that if the ci-framework wants to generate a populated version of this file that the ceph values are known it from here:

https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/cifmw_ceph_client/templates/k8s_ceph_secret.yml.j2

File: examples/va/hci/control-plane.md
Comment (fultonj) [inline examples/va/hci/control-plane.md:41]: The NetConfig generated by this step is missing VLANs.

In main I get this:

```
[zuul@controller-0 architecture]$ kustomize build examples/va/hci/ | grep vlan
      "type": "macvlan",
      "type": "macvlan",
      "type": "macvlan",
      "type": "macvlan",
      vlan: 20
      vlan: 21
      vlan: 22
      vlan: 23
[zuul@controller-0 architecture]$ 
```

In the branch for this PR I get this:
```
[zuul@controller-0 architecture]$ kustomize build examples/va/hci/control-plane | grep vlan
      "type": "macvlan",
      "type": "macvlan",
      "type": "macvlan",
      "type": "macvlan",
[zuul@controller-0 architecture]$ 
```

Comment (abays) [reply examples/va/hci/control-plane.md:41]: Hmm, not sure why that is.  Both branches use the same underlying `lib` stuff for the networking config generation.

Comment (fultonj) [reply examples/va/hci/control-plane.md:41]: I misdiagnosed. It is not related to your patch.

I wasn't testing with the same values file for both of the above outputs. The source from ci-framework is what caused this and it should be fixed by this.

  https://github.com/openstack-k8s-operators/ci-framework/pull/1103/

File: examples/va/hci/edpm-pre-ceph/values.yaml
Comment (pablintino) [inline examples/va/hci/edpm-pre-ceph/values.yaml:21]: I think this one doesn't exist anymore. `timesync_ntp_servers` has replaced it and the structure is slightly different:
```yaml
timesync_ntp_servers:
    - hostname: pool.ntp.org
```

File: examples/va/hci/values.yaml
Comment (cjeanner) [inline examples/va/hci/values.yaml:4]: As discussed, would it be possible to add a new entry in the values.yml `metadata` dict? Something like:
```YAML
metadata:
  name: edpm-values-post-ceph
  label: dataplane (or controlplane, or services [provided we can kustomize that one])
```
Doing so would make automation slightly less complicated, allowing to detect the "type|kind|whatever" we're trying to deploy.

Comment (abays) [reply examples/va/hci/values.yaml:4]: Yes, we can add arbitrary labels using the standard k8s format.  It would look something like so:

```
metadata:
  name: edpm-values-post-ceph
  labels:
    dataplane: true
```

Comment (cjeanner) [reply examples/va/hci/values.yaml:4]: Sounds good! Label names such as `dataplane` and `controlplane`(and, maybe, `service`?) sounds like a good plan - and then I can `metadata.labels.get('dataplane', False)` and I get my "type".

Comment (abays) [reply examples/va/hci/values.yaml:4]: I'm just wondering which particular labels would be appropriate for which `*values.yaml` files.  In the case of the control plane, I think that is pretty obvious.  But this `edpm-values-post-ceph` is a little trickier...maybe.  It has stuff for the `OpenStackDataPlaneNodeSet` but also for a `Secret` and ` ConfigMap` -- but they are all indeed related to the data plane.  I guess it depends on what exactly the labels mean to you.

Could you let me know what labels you would like to see on the various `*values.yaml`s in the PR?  I've listed them below and tried to describe what they customize.

- `control-plane/nncp/values.yaml` (`NetConfig`, `NetworkAttachmentDefinition`, `NNCP`, `L2Advertisement` `IPAddressPool` and `OpenStackControlPlane` customizations for initial control plane)
- `edpm-pre-ceph/values.yaml` (`OpenStackDataPlaneNodeSet` and a `Secret` customizations for pre-Ceph, i.e. initial dataplane)
- `service-values.yaml` (`OpenStackControlPlane` customizations for post-Ceph)
- `values.yaml` (`OpenStackDataPlaneNodeSet`, `Secret` and `ConfigMap` customizations for post-Ceph)

File: examples/va/hci/README.md
Comment (leifmadsen) [reply examples/va/hci/README.md:26]: Feel free to ignore this. Once all the in-flight items are landed, I need to go back and make a cross-repo change for this tiny typo :)

**Review by fultonj — COMMENTED** [review]
Can this be updated to deploy >0 glance instances during the post-ceph stage? 

Without it we get 0 as per 

  https://github.com/openstack-k8s-operators/architecture/pull/78

**Comment by fultonj** [comment]
> Can this be updated to deploy >0 glance instances during the post-ceph stage?
> 
> Without it we get 0 as per
> 
> #78

I see that change has been applied. thanks.

https://github.com/openstack-k8s-operators/architecture/compare/bad5c0011d64ddfe03a3e4b476e61c55ec4208c0..f40aa4904f19b5a5f3c307fbf84ea5c5a64040e7

**Review by fultonj — APPROVED** [review]
Let's merge this today after we get a docs change to accomodate.

**Review by leifmadsen — APPROVED** [review]
Tested this against a CIFMW deployed environment. Was able to create a VM and have everything working. Will work with John to get documentation updated.

**Pull request merged**
